### PR TITLE
[1.11] CVE 2023 0464

### PR DESCRIPTION
**Pull request by jbohanon**
# Description

Bump Alpine dependency in the `kubectl` image for CVE-2023-0464.

Can be tested by running the following commands:
`docker run --rm alpine:3.17.2 apk info libssl3 libcrypto3`
`docker run --rm alpine:3.17.3 apk info libssl3 libcrypto3`

# Checklist:

- [ ] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [ ] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [ ] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [ ] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/8125

### CODE REVIEW

File: jobs/kubectl/Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 FROM bitnami/kubectl:1.22.12 as kubectl
 
-FROM alpine:3.17.2
```
Comment (jenshu) [inline jobs/kubectl/Dockerfile:3]: should we also bump all the other dockerfiles to use 3.17.3?

**Comment by jbohanon** [comment]
/kick [docker push timeout](https://storage.googleapis.com/solo-public-build-logs/logs.html?buildid=6ebad4aa-c11c-4406-8dfb-32ef85b290f5)

**Comment by jbohanon** [comment]
https://github.com/solo-io/gloo/issues/8041

This shows the vulns in all the images

**Comment by bewebi** [comment]
Based on the vulnerabilities listed in #8041, it looks like the CVE may also be found in the `envoy-gloo` image, consumed by the `gloo` and `gloo-envoy-wrapper` images and specified [here](https://github.com/solo-io/gloo/blob/bbb2d34e873dbcdca8ad7e977175a4acd519d46c/Makefile#L53)

We don't seem to have a recent `envoy-gloo` 1.21 release ([this is the latest](https://github.com/solo-io/envoy-gloo/releases/tag/v1.21.5-patch1)) so I'm guessing we don't have a release with the CVE patched
I'm not sure of the support matrix as far as being able to use a more recent minor version of Envoy in Gloo 1.11 (I suspect we can't) so we may need to release `envoy-gloo` 1.21 with the CVE patched and then have Gloo 1.11 consume it in order to fully patch the CVE

I haven't dealt with Gloo CVEs in a while, so if this analysis is off and/or it's still worthwhile to merge this PR while addressing the Envoy piece separately, I'm happy to approve.

**Comment by jbohanon** [comment]
I have confirmed that the new image for `envoy-gloo` contains patched versions of `libssl1.1` and `libcrypto1.1` via docker
```
docker run --rm --entrypoint /bin/sh quay.io/solo-io/envoy-gloo:1.21.5-patch2 -c 'apk info -vv | grep -E "(libssl)|(libcrypto)"'
WARNING: Ignoring https://dl-cdn.alpinelinux.org/alpine/v3.15/main: No such file or directory
WARNING: Ignoring https://dl-cdn.alpinelinux.org/alpine/v3.15/community: No such file or directory
libcrypto1.1-1.1.1t-r3 - Crypto library from openssl
libssl1.1-1.1.1t-r3 - SSL shared libraries
```

**Review by bewebi — DISMISSED** [review]
LGTM

**Comment by jbohanon** [comment]
/kick [[Fail] Generated Kube Code [It] can read and write a gloo resource as a typed kube object ](https://storage.googleapis.com/solo-public-build-logs/logs.html?buildid=989e81f0-1dd7-495c-89bb-b2fecf841055)

**Comment by jbohanon** [comment]
/kick [[Fail] Generated Kube Code [It] can read and write a gloo resource as a typed kube object](https://storage.googleapis.com/solo-public-build-logs/logs.html?buildid=cb065168-00ab-4d8d-980a-97791012d4a5)

**Review by jenshu — DISMISSED** [review]

**Review by jenshu — APPROVED** [review]

**Review by bewebi — APPROVED** [review]

**Pull request merged**